### PR TITLE
Hide blue background placeholder when Home Item Poster Loaded

### LINKED ISSUE #321: Homescreen posters have blue border on older roku
**Describe the bug**
When using an older roku device, posters on the home screen are displaying blue borders.

**To Reproduce**

1. Buy an older roku(3500X in my case)
2. Run the app

**Expected behavior**
App looks and performs the same as newer devices

**Logs**
N/A

**Screenshots**
IMG_20201201_131518

**Additional context**
Bug discovered reviewing #320 
Roku device thats affected:
Model - 3500X(1st gen streaming stick)
Roku OS - 9.4.0
Using latest version of master and latest stable server

### PR DESCRIPTION
**Pull request by neilsb**
Update HomeItem to automatically hide the placeholder blue box and icon when the Poster Image is loaded and displayed.

**Changes**
 * Add function to fire on Image Loading Status Event to hide blue box
 * Ensure Blue Box is made visible again when Item Content changed (to handle recycling of nodes)

**Issues**
fixed #321

### CODE REVIEW

**Review by cewert — CHANGES_REQUESTED** [review]
Go down to one of the "latest in" rows, scroll left(to wrap the screen) and I get his error


```Current Function:
163:  sub focusChanged()
164:  
165:    if m.top.itemHasFocus = true then
166:*     m.itemText.repeatCount = -1
167:    else
168:      m.itemText.repeatCount = 0
169:    end if
170:  
Source Digest(s): 
pkg: dev 1.3.5 f41bd951281da39536aedcefd3e398ba Jellyfin

Invalid value for left-side of expression. (runtime error &he4) in pkg:/components/home/HomeItem.brs(166)```

**Comment by neilsb** [comment]
> Go down to one of the "latest in" rows, scroll left(to wrap the screen) and I get his error

m.ItemText is invalid.  I guess it could be a timing issue in that the focus changing at the same time as the "ItemContent" is being changed (through node recycling), and trying to re-set the m.ItemText property using FindNode().  If it is a timing issue, it's happening very consistently though.

Moving the setting of these field to an init function means they'll only be set once, so if the node is recycled it's also a little more efficient.   I've not been able reproduce the issue after making this change.

File: components/home/HomeItem.brs [hunk 1]
```diff
@@ -178,8 +172,12 @@ end sub
 
 'Hide backdrop and icon when poster loaded
 sub onPosterLoadStatusChanged()
-  if m.itemPoster.loadStatus = "ready" then
+  if m.itemPoster.loadStatus = "ready" and m.itemPoster.uri <> ""  then
+    print m.itemText.text + " image ready - hiding blue"
```
Comment (cewert) [inline components/home/HomeItem.brs]: forgot to remove print statement

**Review by cewert — COMMENTED** [review]
I'm no longer able to reproduce the crash but I'm seeing a new bug. After scrolling around a bunch on a couple "latest in" rows, I'm seeing several movie titles scroll even though they are no longer focused

**Comment by neilsb** [comment]
> I'm no longer able to reproduce the crash but I'm seeing a new bug. After scrolling around a bunch on a couple "latest in" rows, I'm seeing several movie titles scroll even though they are no longer focused

It appears that the `itemHasFocus` property (set automatically by the Roku Component) is not always being set to false (or at least an observer is not firing on its change) when there  a lot of scrolling around.  I've only seen this on the older device currently, and since this is a cosmetic issue whereas there are other changes in this PR that prevent crashing - can we create a new issue for this and get this PR merged?

Just with the Rokus Store Blackout for the festive period happening in a few days we need to get a release published on Monday to let it get through the approval process by the blackout.  I think the fix for the crash (and more visible blue line issue) would be beneficial to get in before this happens.

**Comment by cewert** [comment]
>  can we create a new issue for this and get this PR merged?

That works for me just need to test your new commit

**Comment by neilsb** [comment]
I should have mentioned.  Reason for moving the itemExtraText in there was twofold:

- Missed it when I did the same with the other ones
- When the node was recycled, the "ItemExtraText" was not hidden by default.  The result was if you browsed down and along a few libraries, sometimes when going back up to the My Media top row or libraries, they had some text under the main title (e.g. Episode Names, release years, etc

**Review by cewert — APPROVED** [review]
Blue borders are gone and I can't reproduce the crash

**Pull request merged**